src/packagedcode/golang.py

```diff
@@ -70,10 +71,14 @@ class GolangPackage(models.Package):
 
     @classmethod
     def recognize(cls, location):
-        if fileutils.file_name(location).lower() == 'go.mod':
-            gomod_obj = GoMod()
-            gomod_data = gomod_obj.parse_gomod(location)
+        filename = fileutils.file_name(location).lower()
+        if filename == 'go.mod':
+            _obj = GoMod()
+            gomod_data = _obj.parse_gomod(location)
```

Comment:
Why not make this line `gomod_data = GoMod().parse_gomod(location)`? This would keep the code consistent with the lines below.

Reply:
I thought we change in another PR but we can do this in this PR too :)

Reply:
@majurg I have updated my PR please review it again :)